var/spack/repos/builtin/packages/coreneuron/package.py

```diff
@@ -76,6 +77,11 @@ class Coreneuron(CMakePackage):
     conflicts('+sympy', when='coreneuron@0.17')  # issue with include directories
     conflicts('+ispc', when='~nmodl')
 
+    # raise conflict when trying to install '+gpu' without PGI compiler
+    gpu_compiler_message = "For gpu build use %pgi"
+    conflicts('%gcc', when='+gpu', msg=gpu_compiler_message)
+    conflicts('%intel', when='+gpu', msg=gpu_compiler_message)
```

Comment:
@iomaganaris : I wonder if we should remove this from Spack here. It's ok to use non-pgi compiler as long as  that compiler has support for OpenACC. For example, it was true in case of Cray compiler. Our Cmake has this: https://github.com/BlueBrain/CoreNeuron/blob/02b55506c5c7058a61c66c7143236f4fa181ff26/CMake/OpenAccHelper.cmake#L38

Reply:
for now, lets skip what I said. These are recipes in our BBP Spack and practically we won't be using gcc/intel with GPU.